retry/pubspec.yaml

```diff
@@ -1,5 +1,5 @@
 name: retry
-version: 3.0.1
+version: 3.0.1-null-safety.0
```

Comment:
This should be `4.0.0-nullsafety`

- Packages that aren't pinned by flutter should use a breaking change.
- I think there can't be a `-` in the portion of the version following the first `-`. @jakemac53 do you remember if that is the case?
- The existing published `3.0.1` takes precedence over this one.